numpy/polynomial/chebyshev.py

```diff
@@ -1766,6 +1936,255 @@ def chebfit(x, y, deg, rcond=None, full=False, w=None):
+    full : bool, optional
+        Switch determining nature of return value. When it is False (the
+        default) just the coefficients are returned, when True diagnostic
+        information from the singular value decomposition is also returned.
+    w : array_like, optional
+        Weights. If not None, the contribution of each point
+        ``(x1[i], x2[i], ..., xn[i])`` to the fit is weighted by `w[i]`.
+        Ideally the weights are chosen so that the errors of the products
+        ``w[i]*f[i]`` all have the same variance.  The default value is None.
+
+    Returns
+    -------
+    coef : ndarray, shape (M1, M2,..., Mn)
+        Chebyshev coefficients ordered from low to high.
+    [residuals, rank, singular_values, rcond] : list
+        These values are only returned if `full` = True
+        resid -- sum of squared residuals of the least squares fit
+        rank -- the numerical rank of the scaled Vandermonde matrix
+        sv -- singular values of the scaled Vandermonde matrix
+        rcond -- value of `rcond`.
+        For more details, see `linalg.lstsq`.
+    Warns
+    -----
+    RankWarning
+        The rank of the coefficient matrix in the least-squares fit is
+        deficient. The warning is only raised if `full` = False.  The
+        warnings can be turned off by
+        >>> import warnings
+        >>> warnings.simplefilter('ignore', RankWarning)
+
+    See Also
+    --------
+    chebvalnd, chebgridnd
+
+    Notes
+    -----
+    The solution is the coefficients of the Chebyshev series `p` that
+    minimizes the sum of the weighted squared errors
+    .. math:: E = \\sum_j w_j^2 * |y_j - p(x_j)|^2,
+    where :math:`w_j` are the weights. This problem is solved by setting up
+    as the (typically) overdetermined matrix equation
+    .. math:: V(x, y) * c = w * y,
+    where `V` is the weighted pseudo Vandermonde matrix of `x`, `c` are the
+    coefficients to be solved for, `w` are the weights, and `y` are the
+    observed values.  This equation is then solved using the singular value
+    decomposition of `V`.
+    If some of the singular values of `V` are so small that they are
+    neglected, then a `RankWarning` will be issued. This means that the
+    coefficient values may be poorly determined. Using a lower order fit
+    will usually get rid of the warning.  The `rcond` parameter can also be
+    set to a value smaller than its default, but the resulting fit may be
+    spurious and have large contributions from roundoff error.
+    Fits using Chebyshev series are usually better conditioned than fits
+    using power series, but much can depend on the distribution of the
+    sample points and the smoothness of the data. If the quality of the fit
+    is inadequate splines may be a good alternative.
+
+    References
+    ----------
+    .. [1] Wikipedia, "Curve fitting",
+           http://en.wikipedia.org/wiki/Curve_fitting
+    Examples
+    --------
+    """
```

Comment:
Blank line before `"""`. Not sure if you should have `Examples` if there are none.